wire-ios-request-strategy/Sources/Request Strategies/Connection/ConnectionRequestStrategyTests.swift

```diff
@@ -247,6 +251,59 @@ class ConnectionRequestStrategyTests: MessagingTestBase {
         }
     }
 
+    func testThatOneOnOneResolverIsInvoked_WhenConnectionRequestIsAccepted() throws {
+
+        try syncMOC.performAndWait {
+            // GIVEN
+            let connection = createConnectionPayload(self.oneToOneConnection, status: .accepted)
+            let eventType = try XCTUnwrap(ZMUpdateEvent.eventTypeString(for: Payload.Connection.eventType), "eventType is nil")
+            let eventPayload = Payload.UserConnectionEvent(connection: connection, type: eventType)
+            let payloadData = try XCTUnwrap(eventPayload.payloadData(), "payloadData is nil")
+            let event = updateEvent(from: payloadData)
+
+            mockOneOnOneResolver.resolveOneOnOneConversationWithIn_MockMethod = { _, _ in
+                return OneOnOneConversationResolution.noAction
+            }
+
+            sut.oneOnOneResolutionDelay = 0.3
+
+            // WHEN
+            self.sut.processEvents([event], liveEvents: true, prefetchResult: nil)
+        }
+
+        XCTAssertTrue(waitForAllGroupsToBeEmpty(withTimeout: 0.5))
+
+        // THEN
+        XCTAssertEqual(mockOneOnOneResolver.resolveOneOnOneConversationWithIn_Invocations.count, 1)
+    }
+
+    func testThatOneOnOneResolverIsNotInvokedWithinTimeout_WhenDelayIsLongerThanTimeout() throws {
+
+        let expectation = XCTestExpectation(description: "OneOnOneResolver should not be invoked within the specified timeout")
```

Comment:
I don't see any timeout in this test, so something is hard to understand here.

Reply:
Probably it is required to wait a certain time for the expectation to make this also more robust (otherwise it can be short timing issues).

```swift
func testThatOneOnOneResolverIsNotInvokedWithinTimeout_WhenDelayIsLongerThanTimeout() async throws {
        // GIVEN
        let expectation = self.expectation(description: "OneOnOneResolver should not be invoked within the specified timeout")
        expectation.isInverted = true

        sut.oneOnOneResolutionDelay = 1
        mockOneOnOneResolver.resolveOneOnOneConversationWithIn_MockMethod = { _, _ in
            expectation.fulfill()
            return .noAction
        }

        try await syncMOC.perform {
            // GIVEN
            let connection = self.createConnectionPayload(self.oneToOneConnection, status: .accepted)
            let eventType = try XCTUnwrap(ZMUpdateEvent.eventTypeString(for: Payload.Connection.eventType), "eventType is nil")
            let eventPayload = Payload.UserConnectionEvent(connection: connection, type: eventType)
            let payloadData = try XCTUnwrap(eventPayload.payloadData(), "payloadData is nil")
            let event = self.updateEvent(from: payloadData)

            // WHEN
            self.sut.processEvents([event], liveEvents: true, prefetchResult: nil)
        }

        await fulfillment(of: [expectation], timeout: 0.5)

        XCTAssertEqual(mockOneOnOneResolver.resolveOneOnOneConversationWithIn_Invocations.count, 0, "Expected no invocation due to timeout.")
    }
```

Reply:
Updated the test all together.